keras_cv/visualization/plot_image_gallery.py

```diff
@@ -92,27 +89,88 @@ def unpackage_tfds_inputs(inputs):
             "to be true."
         )
 
-    fig = plt.figure(figsize=(cols * scale, rows * scale))
-    fig.tight_layout()  # Or equivalently,  "plt.tight_layout()"
-    plt.subplots_adjust(wspace=0, hspace=0)
-    plt.margins(x=0, y=0)
-    plt.axis("off")
+    def unpack_images(inputs):
+        return inputs["image"]
+
+    # Calculate appropriate number of rows and columns
+    if rows is None or cols is None:
```

Comment:
If one of them is set and the other isn't, this behaves a bit weirdly. 

If we set `rows=2` and `cols=None`, and our batch size is `9`, we'll end up with a 3x3, which seems counterintuitive.

Reply:
So in this case, we can do something like divide the not-`None` argument such that the batch will create something which at least fixes the user's args. 
```python
if rows is None and cols is not None:
    rows = ceil(batch_size / cols)
elif rows is not None and cols is None:
    cols = ceil(batch_size / rows)
```
This should work in theory.
For your example, it would mean we do `rows=2`, `columns=None`, `batch_size=9` and get `rows=2`, `cols=5`

The final grid would have 5 images in the first row and 4 images in the second row. Is that fine?

Reply:
yeah this sgtm